cloudformation/calculation-athena.yaml

```diff
@@ -217,9 +217,9 @@ Resources:
               COALESCE(invoice.month,txn.month)                     AS month,
               COALESCE(invoice.price,0)-COALESCE(txn.price,0)       AS price_difference,
               COALESCE(invoice.quantity,0)-COALESCE(txn.quantity,0) AS quantity_difference,
-              (CASE WHEN COALESCE(txn.price,0) = 0 then null else 100*(COALESCE(invoice.price,0)-txn.price)/txn.price END)
+              (CASE WHEN COALESCE(txn.price,0) = 0 then (CASE WHEN COALESCE(invoice.price,0) = 0 THEN 0 ELSE null END) ELSE 100*(COALESCE(invoice.price,0)-txn.price)/txn.price END)
```

Comment:
Do you still need the second `COALESCE(invoice.price,0)`? Won't it be certainly non null if we hit that?

Reply:
Yep, agreed...  Will fix.

Reply:
No, actually that's correct as it is -- the first one is an internal clause that only gets evaluated if COALESCE(txn.price,0) = 0.  If that's false, the second one still needs the COALESCE.